lib/webpacker/source.rb

```diff
@@ -1,38 +1,31 @@
-# Translates a logical reference for a pack source into the final path needed in the HTML.
-# This translation takes into account whether digesting is configured to happen, which it
-# is by default in the production environment (as set via
-# `Rails.configuration.x.webpacker[:digesting] = true`).
+require "webpacker/manifest"
+require "webpacker/configuration"
+
+# Translates a logical reference for a pack source into the final
+# path needed in the HTML using generated manifest.json manifest.
 class Webpacker::Source
-  def initialize(name)
-    @name = name
+  class SourceError < StandardError; end
+
+  def initialize(filename)
+    @filename = filename
   end
 
   def path
-    if config[:dev_server_host].present?
-      "#{config[:dev_server_host]}/#{filename}"
-    elsif config[:digesting]
-      File.join(dist_path, digested_filename)
+    if Rails.env.development? && dev_server_enabled?
+      "http://#{dev_server[:host]}:#{dev_server[:port]}/#{filename}"
```

Comment:
Seems like we should extract a new `Webpacker::DevServer` object that can do this work. So it'll just be `if Webpacker::DevServer.running?` and then something like `Webpacker::DevServer.resolve(filename)`.